pydantic_ai_slim/pydantic_ai/_output.py

```diff
@@ -671,7 +759,18 @@ async def process(
 
         if self._function_schema:
             try:
-                output = await self._function_schema.call(output, run_context)
+                # Wraps the output function call in an OpenTelemetry span.
+                if trace_context.call:
+                    span_manager = trace_context.span(trace_context.call, include_tool_call_id=True)
+                else:
+                    function_name = getattr(self._function_schema.function, '__name__', 'output_function')
+                    span_manager = trace_context.span(
+                        _messages.ToolCallPart(tool_name=function_name, args=data), include_tool_call_id=False
+                    )
+                with span_manager as span:
+                    output = await execute_function_call(
+                        self._function_schema, run_context, trace_context, output, span
+                    )
```

Comment:
this logic is still duplicated. `span`, `record_response`, and `execute_function_call` can be combined into one.